File: solutions/csharp/lucians-luscious-lasagna/1/LuciansLusciousLasagna.cs
class Lasagna
{
    public int ExpectedMinutesInOven() {
	return 40;
    }
    
    public int RemainingMinutesInOven(int timeInOven) {
	return ExpectedMinutesInOven() - timeInOven;
    }
    
    public int PreparationTimeInMinutes(int layers) {
	return layers * 2;
    }
    
    public int ElapsedTimeInMinutes(int layers, int timeInOven) {
	return timeInOven + PreparationTimeInMinutes(layers);
    }
}

File: solutions/fsharp/lucians-luscious-lasagna/1/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna


let expectedMinutesInOven = 40

let remainingMinutesInOven ovenTime =
    expectedMinutesInOven - ovenTime

let preparationTimeInMinutes layers = 2 * layers

let elapsedTimeInMinutes layers ovenTime =
    preparationTimeInMinutes layers + ovenTime